exercises/dolica_akelloegwel/ex02_oo_basics/src/Shape.h

```diff
@@ -0,0 +1,31 @@
+#pragma once
+
+#include <string>
+#include <iostream>
+#include <math.h>
+#include <vector>
+#include <algorithm>
+
+class Shape
+{
+protected:
+	double area;
+	double perimeter;
+	std::string type;
+	int numSides;
+	std::string printMessage;
+	void SetPerimeter(double p);
+	void SetArea(double a);
+	void SetSides(int s);
+	std::string CreateBaseMessage();
+	void SetMessage(std::string msg);
+
+public:
+	Shape(std::string t, int n);
+	double GetPerimeter() const;

```

Comment:
Did you consider using virtual methods so that you can reap the benefits of polymorphism? (Talk to me about this if you need more explanation).